pkg/builder/controller_test.go

```diff
@@ -160,6 +161,29 @@ var _ = Describe("application", func() {
 			Expect(instance).NotTo(BeNil())
 		})
 
+		It("should override logger during creation of controller", func() {
+
+			logger := logf.NullLogger{}
+			newController = func(name string, mgr manager.Manager, options controller.Options) (controller.Controller, error) {
+				if options.Log == logger {
+					return controller.New(name, mgr, options)
+				}
+				return nil, fmt.Errorf("logger expected %T but found %T", logger, options.Log)
+			}
+
+			By("creating a controller manager")
+			m, err := manager.New(cfg, manager.Options{})
+			Expect(err).NotTo(HaveOccurred())
+
+			instance, err := ControllerManagedBy(m).
+				For(&appsv1.ReplicaSet{}).
+				Owns(&appsv1.ReplicaSet{}).
+				WithLogger(logger).
+				Build(noop)
+			Expect(err).NotTo(HaveOccurred())
+			Expect(instance).NotTo(BeNil())
```

Comment:
That test will also succeed without your code changes - Please test that you get the same logger as you inserted, not that you get any logger

Reply:
No it does not. I tested it. Without my changes it fails as i get back a `*zap.Logger` In that case.

Reply:
While the test was working I made it more explicit by using a minimal custom `testLogger`.
Without my changes the test now fails with:
```
logger expected *builder.testLogger but found *zapr.zapLogger
```